File: python/cluster_management/src/update_cluster.py
import boto3
import os


def update_cluster(region, cluster_id, deletion_protection_enabled):
    try:
        client = boto3.client("dsql", region_name=region)
        return client.update_cluster(identifier=cluster_id, deletionProtectionEnabled=deletion_protection_enabled)
    except:
        print("Unable to update cluster")
        raise


def main():
    region = os.environ.get("CLUSTER_REGION", "us-east-1")
    cluster_id = os.environ.get("CLUSTER_ID")
    assert cluster_id is not None, "Must provide CLUSTER_ID"
    deletion_protection_enabled = False
    response = update_cluster(region, cluster_id, deletion_protection_enabled)
    print(f"Updated {response['arn']} with deletion_protection_enabled: {deletion_protection_enabled}")


if __name__ == "__main__":
    main()

File: javascript/cluster_management/src/update_cluster.js
import { DSQLClient, UpdateClusterCommand } from "@aws-sdk/client-dsql";

export async function updateCluster(region, clusterId, deletionProtectionEnabled) {

  const client = new DSQLClient({ region });

  const updateClusterCommand = new UpdateClusterCommand({
    identifier: clusterId,
    deletionProtectionEnabled: deletionProtectionEnabled
  });

  try {
    return await client.send(updateClusterCommand);
  } catch (error) {
    console.error("Unable to update cluster", error.message);
    throw error;
  }
}

async function main() {
  const region = process.env.CLUSTER_REGION || "us-east-1";
  const clusterId = process.env.CLUSTER_ID;

  if (!clusterId) {
    console.error("Error: CLUSTER_ID environment variables must be set");
    process.exit(1);
  }

  const response = await updateCluster(region, clusterId, false);
  console.log(`Updated ${response.arn}`);
}

if (process.env.NODE_ENV !== 'test') {
  main();
}
